src/pages/api/import-exhibitor.ts

```diff
@@ -0,0 +1,128 @@
+import { z } from "zod";
+import { env } from "@/env.mjs";
+import { prisma } from "@/server/db";
+import { timingSafeEqual, randomUUID } from "crypto";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { validateOrganizationNumber } from "@/shared/validateOrganizationNumber";
+
+const importToken = Buffer.from(env.IMPORT_TOKEN);
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const authHeader = req.headers["authorization"];
+  if (req.method !== "PUT") return res.status(405).end();
+  else if (
+    authHeader == undefined ||
+    authHeader.length != importToken.length ||
+    !timingSafeEqual(Buffer.from(authHeader), importToken)
+  ) {
+    return res.status(402).end();
+  }
+
+  const bodySchema = z.object({
+    contactPerson: z.string(),
+    telephoneNumber: z.string(),
+    companyName: z.string(),
+    organizationNumber: z.string().transform((s, ctx) => {
+      const orgNum = validateOrganizationNumber(s);
+      if ("error" in orgNum) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: orgNum.error,
+        });
+        return z.NEVER;
+      }
+      return orgNum.value;
+    }),
+    email: z.string(),
+    exhibitorPackage: z.enum(["base", "sponsor", "headhunter", "premium"]),
+  });
+
+  const body = bodySchema.safeParse(req.body);
+  if (!body.success) return res.status(422).end(body.error);
+
+  const {
+    contactPerson,
+    telephoneNumber,
+    companyName,
+    organizationNumber,
+    email,
+    exhibitorPackage,
+  } = body.data;
+
+  const exhibitor = await prisma.exhibitor.findUnique({
```

Comment:
The `prisma.exhibitor.upsert` returns the exhibitor that we created/updated so we can use the `id` in the return value from that when upserting the user